tools/mdLint.ts

```diff
@@ -0,0 +1,38 @@
+import fs from "fs";
+import path from "path";
+import { exec } from "../src/lib/shell";
+
+// get sub folders under commands folder.
+const getMarkdownFiles = (curDir: string, mds: string[]): void => {
+  fs.readdirSync(curDir).forEach((f) => {
+    const p = path.join(curDir, f);
+    if (fs.lstatSync(p).isDirectory()) {
+      getMarkdownFiles(p, mds);
+    } else if (p.endsWith(".md")) {
+      mds.push(p);
+    }
+  });
+};
+
+const testMd = async (f: string): Promise<void> => {
+  try {
+    await exec("./node_modules/.bin/markdown-link-check", [f]);
+  } catch (e) {
+    console.log(`${f}: ${e.message}`);
+    process.exit(1);
+  }
+};
+
+(async (): Promise<void> => {
```

Comment:
I think this script file may be un-necessary:

See: https://github.com/tcort/markdown-link-check#check-links-from-a-local-markdown-folder-recursive 

You can inline this into a package.json script

Reply:
i cannot because i need a status code 1 to stop the script. :-)
the inline approach always return 0

Reply:
Sorry Dennis, not quite sure I understand what you're saying. Npm scripts will exit with 1 if if any of the commands fail:

```json
  "scripts": {
    "i-will-fail": "false && true"
  },
```

```sh
npm run i-will-fail # will exit with code 1
```

are you thinking othewise?

Reply:
maybe you can try it yourself and let me know what you get
add this to your package.json
```
  "md-lint": "find guides -name \\*.md -exec markdown-link-check {} \\",
```
or just ran (at the root of spk repo)
```
find guides -name \*.md -exec markdown-link-check {} \;
```
then 
```
echo $?
```
there are errors reported by the tool and it is returning 0 status code.

Reply:
@dennisseah try the suggestion here: https://github.com/tcort/markdown-link-check/issues/57#issuecomment-407061525

```
find guides -name \\*.md | xargs -n 1 markdown-link-check
```

Reply:
@bnookala 
and what about people using windows?

Reply:
@dennisseah this isn't a run-time requirement - just a development requirement. I expect that developers who are running windows who also want to contribute to spk development should do their work in a WSL environment.

Reply:
ok then. We do not support windows runtime.